### PR TITLE
jayanath/c04-iac01

### PR DESCRIPTION
**Pull request by jayanath**
# Description

Exercise (e.g. C01-E01): c04-iac01

Student Name: Jayanath Amaranayake

> A brief explanation of what has been done.
Add Terraform code for EC2 and S3 tasks

## Type of change

- [x] Exercise Submission (one exercise per PR)
  - [x] This PR is just for **one** exercise of a class
  - [x] PR name follows the pattern `<github-username>/<exercise-number>`
  - [x] Added exercise files inside folder `/classes/<class- number>/exercises/<exercise-number>/<github-username>/`
	- [ ] Review [exercise submission steps](./README.md#Exercises)
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

### CODE REVIEW

File: classes/04class/exercises/c04-iac01/jayanath/_backend.tf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+terraform {
+  backend "s3" {
+    bucket = "devops-academy-tf-backends-9930.com"
+    key    = "exercises/c04-iac01"
+    region = "ap-southeast-2"
+  }
```
Comment (alanlima) [inline classes/04class/exercises/c04-iac01/jayanath/_backend.tf:6]: Nice, using s3 backend 👌

Comment (jayanath) [reply classes/04class/exercises/c04-iac01/jayanath/_backend.tf:6]: Thanks @alanlima 🙇‍♂️

File: classes/04class/exercises/c04-iac01/jayanath/ec2.tf [hunk 2]
```diff
@@ -0,0 +1,19 @@
+data "aws_ami" "amazon-linux-2" {
+  most_recent = true
+  owners      = ["amazon"]
+
+  filter {
+    name   = "name"
+    values = ["amzn2-ami-hvm*"]
+  }
+}
```
Comment (denstorti) [inline classes/04class/exercises/c04-iac01/jayanath/ec2.tf:9]: 👍

File: classes/04class/exercises/c04-iac01/jayanath/iam.tf [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        "Action": "sts:AssumeRole",
+        "Principal": {
+          "Service": "ec2.amazonaws.com"
+        },
+        "Effect": "Allow",
+        "Sid": ""
+      }
+    ]
+  }
+  EOF
```
Comment (denstorti) [inline classes/04class/exercises/c04-iac01/jayanath/iam.tf:19]: if this gets big, an option is template files: https://medium.com/@mitesh_shamra/terraform-managing-policy-using-templates-e254b2d17907

**Pull request merged**